src/Compilers/Core/MSBuildTask/Microsoft.Managed.Core.targets

```
@@ -256,12 +256,41 @@
     If a user requests that any @(AdditionalFiles) items are copied to the output directory
     we add them to the @(None) group to ensure they will be copied.
  -->
- 
-   <Target Name="CopyAdditionalFiles" 
-           BeforeTargets="AssignTargetPaths">
+
+  <Target Name="CopyAdditionalFiles"
+          BeforeTargets="AssignTargetPaths">
     <ItemGroup>
       <None Include="@(AdditionalFiles)" Condition="'%(AdditionalFiles.CopyToOutputDirectory)' != ''" />
     </ItemGroup>
   </Target>
 
+  <!--
+    ========================
+    GeneratedFilesOutputPath
+    ========================
+    
+    Controls output of generated files.
+    
+    GeneratedFilesOutputPath controls the location the files will be output to.
+    The compiler will not emit any generated files when the path is empty, and defaults to a /generated directory in $(IntermediateOutputPath) if $(IntermediateOutputPath) has a value.
+    EmitGeneratedFiles allows the user to control if anything is emitted by clearing the property when not true.
+    When GeneratedFilesOutputPath is set we ensure the directory exists, and issue a warning if EmitGeneratedFiles is true, but no GeneratedFilesOutputPath is set
+    -->
+  <PropertyGroup>
+    <EmitGeneratedFiles Condition="'$(EmitGeneratedFiles)' == ''">true</EmitGeneratedFiles>
```

I'm much more concerned about the _default_ value here than I am the feature as a whole.

```suggestion
    <EmitGeneratedFiles Condition="'$(EmitGeneratedFiles)' == ''">false</EmitGeneratedFiles>
```

---

Echoing this a bit: I absolutely agree this is a feature that _should_ exist, but I'd hope it's not something people are needing to use regularly. Now that we have the IDE making it possible to browse to files and will start to provide a superior experience than what this could provide, have we met the primary customer need there?